modules/src/main/assembly/dist.xml

```diff
@@ -140,6 +145,15 @@
             <unpack>false</unpack>
             <includes>
                 <include>org.hibernate:hibernate-search-elasticsearch</include>
+            </includes>
+        </dependencySet>
+
+        <dependencySet>
+            <useProjectArtifact>false</useProjectArtifact>
+            <outputDirectory>org/hibernate/search/elasticsearch-client/${hibernate.search.version}</outputDirectory>
+            <useTransitiveFiltering>false</useTransitiveFiltering>
+            <unpack>false</unpack>
+            <includes>
                 <include>com.google.code.gson:gson</include>
```

Comment:
The rest client doesn't depend on gson (unlike Jest), so this should be moved to `org.hibernate:hibernate-search-elasticsearch` just above.